rdr_service/offline/study_nph_biobank_file_export.py

```diff
@@ -55,24 +55,18 @@ def _get_nph_participant(participant_id: int) -> NphParticipant:
         return session.query(NphParticipant).get(participant_id)
 
 
-def _get_latest_biobank_file_export() -> BiobankFileExport:
-    nph_biobank_file_export = NphBiobankFileExportDao()
-    with nph_biobank_file_export.session() as session:
-        return session.query(BiobankFileExport).order_by(
-            BiobankFileExport.created.desc()
-        ).first()
-
-
-def _get_sample_updates_since_last_file_export() -> Iterable[SampleUpdate]:
-    last_successful_biobank_file_export = _get_latest_biobank_file_export()
-    if last_successful_biobank_file_export is not None:
-        last_successful_biobank_file_export_date = last_successful_biobank_file_export.created
-    else:
-        last_successful_biobank_file_export_date = datetime(year=2023, month=1, day=1)
+def _get_sample_updates_needing_export() -> Iterable[SampleUpdate]:
+    """
+    Return any SampleUpdates that haven't already been included in an export
+    """
     sample_update_dao = NphSampleUpdateDao()
     with sample_update_dao.session() as session:
-        return session.query(SampleUpdate).filter(
-            SampleUpdate.created >= last_successful_biobank_file_export_date
+        return session.query(
+            SampleUpdate
+        ).outerjoin(
+            SampleExport
```

Comment:
SQLALchemy infers what to join to for this? TIL

Reply:
Yeah, as long as there's only one other model that relates to what's being joined, and it only relates through one foreign key, then SQLAlchemy will essentially assume you meant the one and only possible thing.